### DISCUSSION TITLE
VPN and connection drops

### DISCUSSION THREAD

**Discussion by nata1993** [TeamNewPipe/NewPipe#10679] [Questions] [answered]
Hello! I am writing to you with regards of using VPN with Newpipe running at the same time. I have encountered peculiar situation regarding VPN usage, VPN provider is Mullvad, where in very specific situation Newpipe is having a trouble of keeping connection or getting packets during communication. I do not have any issues when stationary. I do however have issues when driving a car, putting Newpipe to play music on the background (phone screen off) while VPN is running. The feed plays for a while but when next music title starts buffering, it takes a lot of time, up to 30 seconds, before next title starts playing. After this, it gets even worse. The buffering starts in the middle of the title. For now I have created split tunneling in VPN where Newpipe connection is not goung through the VPN anymore and the issues are gone. Afterall, while not stationary, the cellular network always changes with each new cellular mast connection. However, the question still remains, does Newpipe just dislike VPN connections for whatever reason there could be or its more of the network side issue and VPN fault?

**Comment by opusforlife2** [comment]
What about other apps? Youtube, Libretube, Piped in the browser, etc.

**Comment by doobydooby72** [comment, marked as answer]
I had a similar issue a while ago and also of recent. Seems to be all yt apps with my offical vpn vendors app. I ended up downloading the wireguard app, got the config file from my vpn and just loaded that in and it hasnt been a problem for me since.

I do use a different vpn to yours but it seemed to be an issue specifically with the offical vpn app and any youtube app on multiple devices and even though the official app has features i like it obviously not the vpn service itself since their wireguard and openvpn configs worked fine with both of those protocols official apps instead (havent tried openvpn in ages)
